core/src/test/java/org/apache/iceberg/avro/TestAvroNameMapping.java

```diff
@@ -80,10 +80,7 @@ public void testMapProjections() throws IOException {
 
     Record projected = writeAndRead(writeSchema, readSchema, record, nameMapping);
     // field id 5 comes from read schema
-    Assertions.assertThat(projected.getSchema().getField("location_r5"))
-        .as("Field missing from table mapping is renamed")
-        .isNotNull();
-    Assertions.assertThat(projected.get("location_r5"))
+    Assertions.assertThat(projected.get("location"))
```

Comment:
These test updates fix the odd behavior caused by rewriting the read schema. Name mapping previously needed to produce fields with incorrect names in order to project fields but not have them read (by name) from the data file.